### PR TITLE
Feature/calypso reader design

### PR DESCRIPTION
**Pull request by nbradbury**
Addresses #3037 & #3040 - redesigns the reader list & detail screens to more closely match the Calypso reader.

reader-card

### CODE REVIEW

**Comment by tonyr59h** [comment]
Looking good! Just getting through with a first-pass review, I'll delve into the code next.

* Scrollbar on Reader detail screen extends into the ActionBar at the top and Comment/Like bar at the bottom
* The Like star animation is clipped by the parent container
* When likes and comments are unavailable the Reader detail still shows the bar, but it's empty. [See here.](https://cldup.com/CwokPS3ceg.png)
* Here's something weird that might just be an Android bug. Highlight a word in an article on the detail view then click a link. Press Back to return to the article. The word and selected link will remain highlighted until you back out of the detail screen. Definitely not a priority to fix, just odd. [See here.](https://cldup.com/UDScd34_Kk.png)
* The hit region on the '...' item seems unnecessarily small, I frequently mis-click and end up selecting the article header. Could we [expand it](https://cldup.com/P1bWruEuxW.png) with padding?

> Blog name replaces tag name in title

* My title bar is just showing 'Reader', no blog name or tag name

> Move view & share icons to "..." menu

* Is this something we still want to do in the Reader detail?

**Comment by nbradbury** [comment]
> Scrollbar on Reader detail screen extends into the ActionBar at the top and Comment/Like bar at the bottom

I'm not sure I can do anything about that since the scrollview containing the content has white space at the top which matches the height of the toolbar (otherwise the toolbar would overlay the content). I did improve this a little bit when scrolling down in 21827ec by not hiding the toolbar until the user scrolls more than the height of the toolbar, but it's still pretty obvious when scrolling back up.

> The Like star animation is clipped by the parent container

Fixed in cca044b.

> When likes and comments are unavailable the Reader detail still shows the bar

Nice catch! Fixed in 31d0220.

> Here's something weird that might just be an Android bug...

Wow, that's definitely weird. I think you're right that it's an Android bug, though - that whole selection CAB is handled entirely by the system.

> The hit region on the '...' item seems unnecessarily small

Fixed in 02c74c2.

> "Blog name replaces tag name in title" - My title bar is just showing 'Reader', no blog name or tag name

Showing "Reader" consistently was a recent design change (I should've updated the issue to reflect this).

> "Move view & share icons to '...' menu" - Is this something we still want to do in the Reader detail?

The design for this is changing and isn't complete - let's revisit this when you're ready to merge.

**Comment by tonyr59h** [comment]
Just noticed that the PTR view is drawn under the tag spinner view. You can only see it with long tag names like `Business & Technology`.

**Comment by tonyr59h** [comment]
I opened up three issues for things we discussed previously. Don't think I can satisfactorily finish the code review now, got back a bit later than expected and I'm exhausted so I'm going to push it off till first thing tomorrow.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.java [hunk 1]
```diff
@@ -934,4 +829,61 @@ void pauseWebView() {
...
+     * can we show the footer bar which contains the like & comment counts?
+     */
+    private boolean canShowFooter() {
+        return canShowLikeCount() || canShowCommentCount();
+    }
+
+    private boolean canShowCommentCount() {
+        return mPost != null
+                && mPost.isWP()
+                && !mPost.isJetpack
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.java]: If the post `isWP()` does that imply `!isJetpack`?

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListActivity.java [hunk 2]
```diff
@@ -106,14 +122,96 @@ public void onBackPressed() {
...
+        }
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
         if (item.getItemId() == android.R.id.home) {
             onBackPressed();
             return true;
+        } else if (mFollowMenuItem != null && item.equals(mFollowMenuItem)) {
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListActivity.java]: nitpick; no need to check if mFollowMenuItem here, `.equals` should take care of that.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java [hunk 3]
```diff
@@ -194,13 +224,30 @@ public void onClick(View view) {
...
+            postHolder.txtWordCount.setVisibility(View.VISIBLE);
+        } else {
+            postHolder.txtWordCount.setVisibility(View.GONE);
+        }
+
         // show the best tag for this post
         final String tagToDisplay = (mCurrentTag != null ? post.getTagForDisplay(mCurrentTag.getTagName()) : null);
         if (!TextUtils.isEmpty(tagToDisplay)) {
-            postHolder.txtTag.setText(tagToDisplay);
+            if (tagToDisplay.startsWith("#")) {
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java]: This is the third place I've seen this sort of logic. Maybe worth making a static method for it?

File: WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java [hunk 4]
```diff
@@ -94,6 +95,21 @@ public static String getLongLikeLabelText(Context context, int numLikes, boolean
     }
 
     /*
+     * short like text ("1 like," "5 likes," etc.)
+     */
+    public static String getShortLikeLabelText(Context context, int numLikes) {
```
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java]: nitpick; could use `plurals` here.

File: WordPress/src/main/res/anim/activity_slide_in_from_right.xml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 
 <set xmlns:android="http://schemas.android.com/apk/res/android"
-    android:duration="@integer/reader_activity_anim_time"
+    android:duration="@android:integer/config_shortAnimTime"
```
Comment (tonyr59h) [inline WordPress/src/main/res/anim/activity_slide_in_from_right.xml]: :+1:

File: WordPress/src/main/res/drawable/reader_button_comment.xml [hunk 6]
```diff
@@ -1,9 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <selector xmlns:android="http://schemas.android.com/apk/res/android">
-
-    <item android:drawable="@drawable/ic_comment_disabled" android:state_enabled="false" />
-    <item android:drawable="@drawable/ic_comment_active" android:state_pressed="true" />
-    <item android:drawable="@drawable/ic_comment_active" android:state_selected="true" />
-    <item android:drawable="@drawable/ic_comment" />
-
+    <item android:drawable="@drawable/reader_comment_active" android:state_pressed="true" />
```
Comment (tonyr59h) [inline WordPress/src/main/res/drawable/reader_button_comment.xml]: No longer showing a disabled state?

File: WordPress/src/main/res/layout/reader_follow_button.xml [hunk 7]
```diff
@@ -4,21 +4,29 @@
...
     android:paddingBottom="@dimen/margin_small"
-    android:paddingLeft="@dimen/margin_medium"
-    android:paddingRight="@dimen/margin_medium"
+    android:paddingLeft="@dimen/margin_small"
+    android:paddingRight="@dimen/margin_small"
     android:paddingTop="@dimen/margin_small">
 
+    <ImageView
+        android:id="@+id/image_follow_button"
+        android:layout_width="16dp"
```
Comment (tonyr59h) [inline WordPress/src/main/res/layout/reader_follow_button.xml]: nitpick; use dimen resources.

**Comment by tonyr59h** [comment]
There's some funky offline behavior I noticed with the Like button. Can we disable it when there's no connection?

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.java (hunk 1 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.java]: > If the post isWP() does that imply !isJetpack?

That used to be the case, but not any more. The short explanation is that there are two sources for the Reader: WordPress blogs and external RSS feeds. In the past Jetpack blogs were treated like RSS, but recent changes have started treating them like WP blogs so we can poll them more efficiently and support likes. Comments are still a WIP.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java (hunk 4 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java]: I wasn't aware of [plurals](http://developer.android.com/guide/topics/resources/string-resource.html#Plurals) until now! I don't see us using plurals anywhere at the moment - do we know whether we can integrate them into our translation process?

File: WordPress/src/main/res/drawable/reader_button_comment.xml (hunk 6 above)
Comment (nbradbury) [inline WordPress/src/main/res/drawable/reader_button_comment.xml]: I got rid of the disabled state for two reasons: one, it's no longer used; and two, the normal color is already grey.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java (hunk 3 above)
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java]: Good call: you caught me being lazy :) Added in e12a076.

**Comment by nbradbury** [comment]
> There's some funky offline behavior I noticed with the Like button.

Resolved in 4db98ea.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java (hunk 4 above)
Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java]: That's a good question, I'll have to ask i18n or do some testing on Glotpress.

**Comment by tonyr59h** [comment]
Gonna test out the latest commits then I think it's good to merge. Great work! :tada:

**Comment by tonyr59h** [comment]
:shipit:

**Pull request merged**